<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AudioRecorder</class>
 <widget class="QMainWindow" name="AudioRecorder">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>398</width>
    <height>565</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="3" column="0">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QTabWidget" name="tabWidget">
        <property name="currentIndex">
         <number>1</number>
        </property>
        <widget class="QWidget" name="tab">
         <attribute name="title">
          <string>Play/Record</string>
         </attribute>
         <widget class="QWidget" name="layoutWidget">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>10</y>
            <width>381</width>
            <height>322</height>
           </rect>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="4" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Sample rate:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="audioDeviceBox"/>
           </item>
           <item row="3" column="1">
            <widget class="QComboBox" name="containerBox"/>
           </item>
           <item row="4" column="1">
            <widget class="QComboBox" name="sampleRateBox"/>
           </item>
           <item row="2" column="1">
            <widget class="QComboBox" name="audioCodecBox"/>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>File Container:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Audio Codec:</string>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Channels:</string>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QComboBox" name="channelsBox"/>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Input Device:</string>
             </property>
            </widget>
           </item>
           <item row="6" column="0" colspan="2">
            <widget class="QGroupBox" name="groupBox">
             <property name="title">
              <string>Encoding Mode:</string>
             </property>
             <layout class="QGridLayout" name="gridLayout">
              <item row="0" column="0" colspan="2">
               <widget class="QRadioButton" name="constantQualityRadioButton">
                <property name="text">
                 <string>Constant Quality:</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <spacer name="horizontalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeType">
                 <enum>QSizePolicy::Fixed</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="1" column="1">
               <widget class="QSlider" name="qualitySlider">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item row="2" column="0" colspan="2">
               <widget class="QRadioButton" name="constantBitrateRadioButton">
                <property name="text">
                 <string>Constant Bitrate:</string>
                </property>
               </widget>
              </item>
              <item row="3" column="0">
               <spacer name="horizontalSpacer">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeType">
                 <enum>QSizePolicy::Fixed</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item row="3" column="1">
               <widget class="QComboBox" name="bitrateBox">
                <property name="enabled">
                 <bool>false</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
        <widget class="QWidget" name="cepstrumTab">
         <attribute name="title">
          <string>Cepstrum and Neural Network</string>
         </attribute>
         <widget class="QPushButton" name="calculateCepstrumButton">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>10</y>
            <width>161</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>Calculate Cepstrum for record</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_6">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>100</y>
            <width>101</width>
            <height>17</height>
           </rect>
          </property>
          <property name="text">
           <string>Detected word:</string>
          </property>
         </widget>
         <widget class="QLabel" name="detectedNumberLabel">
          <property name="geometry">
           <rect>
            <x>80</x>
            <y>100</y>
            <width>111</width>
            <height>17</height>
           </rect>
          </property>
          <property name="text">
           <string>None</string>
          </property>
         </widget>
         <widget class="QPushButton" name="outputButton">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>120</y>
            <width>81</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>Output...</string>
          </property>
         </widget>
         <widget class="QPushButton" name="recordButton">
          <property name="geometry">
           <rect>
            <x>104</x>
            <y>120</y>
            <width>89</width>
            <height>20</height>
           </rect>
          </property>
          <property name="text">
           <string>Record</string>
          </property>
         </widget>
         <widget class="QLabel" name="levelLabel">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>168</y>
            <width>89</width>
            <height>17</height>
           </rect>
          </property>
          <property name="text">
           <string>Audio Level:</string>
          </property>
         </widget>
         <widget class="QPushButton" name="playButton">
          <property name="geometry">
           <rect>
            <x>11</x>
            <y>139</y>
            <width>81</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>Play</string>
          </property>
         </widget>
         <widget class="QWidget" name="layoutWidget_2">
          <property name="geometry">
           <rect>
            <x>100</x>
            <y>168</y>
            <width>185</width>
            <height>17</height>
           </rect>
          </property>
          <layout class="QVBoxLayout" name="levelsLayout1"/>
         </widget>
         <widget class="QPushButton" name="pauseButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="geometry">
           <rect>
            <x>105</x>
            <y>139</y>
            <width>89</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>Pause</string>
          </property>
         </widget>
         <widget class="QPushButton" name="locateRecordToAnalyzeButton">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>70</y>
            <width>161</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>Locate record to analyze...</string>
          </property>
         </widget>
         <widget class="QPushButton" name="runNeuralNetworkButton">
          <property name="geometry">
           <rect>
            <x>10</x>
            <y>40</y>
            <width>161</width>
            <height>21</height>
           </rect>
          </property>
          <property name="text">
           <string>Run neural network</string>
          </property>
         </widget>
         <widget class="QComboBox" name="regulatorComboBox">
          <property name="geometry">
           <rect>
            <x>280</x>
            <y>110</y>
            <width>72</width>
            <height>22</height>
           </rect>
          </property>
          <item>
           <property name="text">
            <string>P</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>PI</string>
           </property>
          </item>
         </widget>
         <widget class="QCustomPlot" name="Plot1" native="true">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>210</y>
            <width>331</width>
            <height>271</height>
           </rect>
          </property>
         </widget>
         <widget class="QPushButton" name="stopButton">
          <property name="geometry">
           <rect>
            <x>280</x>
            <y>70</y>
            <width>81</width>
            <height>26</height>
           </rect>
          </property>
          <property name="text">
           <string>Stop</string>
          </property>
         </widget>
         <widget class="QPushButton" name="stepButton">
          <property name="geometry">
           <rect>
            <x>280</x>
            <y>40</y>
            <width>81</width>
            <height>26</height>
           </rect>
          </property>
          <property name="text">
           <string>Step</string>
          </property>
         </widget>
         <widget class="QPushButton" name="startButton">
          <property name="geometry">
           <rect>
            <x>280</x>
            <y>10</y>
            <width>81</width>
            <height>26</height>
           </rect>
          </property>
          <property name="text">
           <string>Start</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_8">
          <property name="geometry">
           <rect>
            <x>190</x>
            <y>45</y>
            <width>41</width>
            <height>17</height>
           </rect>
          </property>
          <property name="text">
           <string>SetPoint</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_9">
          <property name="geometry">
           <rect>
            <x>210</x>
            <y>75</y>
            <width>21</width>
            <height>17</height>
           </rect>
          </property>
          <property name="text">
           <string>Gain</string>
          </property>
         </widget>
         <widget class="QSlider" name="horizontalSlider">
          <property name="geometry">
           <rect>
            <x>220</x>
            <y>10</y>
            <width>51</width>
            <height>18</height>
           </rect>
          </property>
          <property name="minimum">
           <number>20</number>
          </property>
          <property name="maximum">
           <number>2000</number>
          </property>
          <property name="singleStep">
           <number>100</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
         <widget class="QSpinBox" name="spinBoxGain">
          <property name="geometry">
           <rect>
            <x>230</x>
            <y>70</y>
            <width>48</width>
            <height>26</height>
           </rect>
          </property>
          <property name="maximum">
           <number>10</number>
          </property>
          <property name="value">
           <number>2</number>
          </property>
         </widget>
         <widget class="QSpinBox" name="spinBoxSP">
          <property name="geometry">
           <rect>
            <x>230</x>
            <y>40</y>
            <width>48</width>
            <height>26</height>
           </rect>
          </property>
          <property name="value">
           <number>10</number>
          </property>
         </widget>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>recordButton</sender>
   <signal>clicked()</signal>
   <receiver>AudioRecorder</receiver>
   <slot>toggleRecord()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>342</x>
     <y>137</y>
    </hint>
    <hint type="destinationlabel">
     <x>221</x>
     <y>264</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>constantQualityRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>qualitySlider</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>113</x>
     <y>197</y>
    </hint>
    <hint type="destinationlabel">
     <x>115</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pauseButton</sender>
   <signal>clicked()</signal>
   <receiver>AudioRecorder</receiver>
   <slot>togglePause()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>438</x>
     <y>137</y>
    </hint>
    <hint type="destinationlabel">
     <x>221</x>
     <y>264</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>outputButton</sender>
   <signal>clicked()</signal>
   <receiver>AudioRecorder</receiver>
   <slot>setOutputLocation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>246</x>
     <y>137</y>
    </hint>
    <hint type="destinationlabel">
     <x>221</x>
     <y>264</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>constantBitrateRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>bitrateBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>173</x>
     <y>259</y>
    </hint>
    <hint type="destinationlabel">
     <x>190</x>
     <y>291</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setOutputLocation()</slot>
  <slot>toggleRecord()</slot>
  <slot>togglePause()</slot>
 </slots>
</ui>
